clarity/src/vm/types/signatures.rs

```diff
@@ -491,10 +501,59 @@ impl TypeSignature {
                     false
                 }
             }
+            PrincipalType => {
+                if other == &PrincipalType {
+                    true
+                } else if let CallableType(CallableSubtype::Principal(_)) = other {
+                    true
+                } else {
+                    false
+                }
+            }
             NoType => panic!("NoType should never be asked to admit."),
             _ => other == self,
         }
     }
+
+    pub fn concretize(&self) -> Result<TypeSignature> {
```

Comment:
Can you add a docstring to this? Thanks!

Reply:
2f083c66b